src/common/modal.js

```diff
@@ -133,6 +134,89 @@ export const confirm = ({
   })
 }
 
+@propTypes({
+  children: propTypes.node.isRequired,
+  icon: propTypes.string,
+  items: propTypes.object.isRequired,
+  title: propTypes.node.isRequired
+})
+class ChooseAction extends Component {
+  _resolve = val => {
+    this.props.resolve(val)
+    instance.close()
+  }
+
+  _reject = () => {
+    this.props.reject()
+    instance.close()
+  }
+
+  render () {
+    const { Body, Footer, Header, Title } = ReactModal
+
+    const {
+      icon,
+      items,
+      title
+    } = this.props
+
+    const body = _addRef(this.props.children, 'body')
+
+    return <div>
+      <Header closeButton>
+        <Title>
+          {icon !== undefined
+            ? <span><Icon icon={icon} /> {title}</span>
+            : title
+          }
+        </Title>
+      </Header>
+      <Body>
+        {body}
+      </Body>
+      <Footer>
+        {map(
+          items,
+          item => <span>
+            <Button
+              className={item.style !== undefined && item.style}
+              onClick={() => this._resolve(item.value)}
+            >
+              {item.name}
+            </Button>
+            {' '}
+          </span>
+        )}
+        <Button onClick={this._reject} >
+          {_('chooseCancel')}
+        </Button>
+      </Footer>
+    </div>
+  }
+}
+
+export const chooseAction = ({
```

Comment:
I called him choose at first and i changed to chooseAction to be more explicit because its role is choosing an action.